ipq40xx: add 5.10 kernel support

robimarko opened the pull request:
OpenWrt has had the 5.10 support for a while, so let's add support for 5.10 under the popular ipq40xx target.

I have split the changes into multiple commits to make it possible to track changes down later if a regression
occurs.

Note that this is not based on PR #4134

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko committed:
ipq40xx: 5.10: drop upstreamed patches

Drop patches that have been upstreamed in before 5.10.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko committed:
ipq40xx: 5.10: remove duplicate GPIO export patch

Its in the generic target already, so remove the duplicate
as it breaks compilation.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko committed:
ipq40xx: 5.10: replace patches with upstreamed versions

USB and SDHCI LDO DTS patches have been upstreamed into 5.12, so
replace the local versions with upstreamed ones.

Reorder, and clearly mark the kernel version.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko committed:
ipq40xx: 5.10: refresh patches to apply

Refresh the kernel patches on top of 5.10 so they apply.
Manually fixup the 705-net-add-qualcomm-ar40xx-phy.patch
to apply.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko committed:
ipq40xx: 5.10: add kernel config

Add config for the 5.10 kernel.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko committed:
ipq40xx: add testing support for kernel 5.10

Add kernel 5.10 as the testing kernel to ipq40xx to
get wider testing.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko committed:
ipq40xx: net: ethernet: edma: update of_get_phy_mode() for 5.10

In kernel v5.5 of_get_phy_mode had its API changed, so its now returning 0
or errors instead of phymode.

Phymode is now returning by passing a pointer to phy_interface_t where it
will be stored.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko committed:
ipq40xx: net: ethernet: edma: reject unsupported coalescing params

Set ethtool_ops->supported_coalesce_params to let
the core reject unsupported coalescing parameters.

This driver did not previously reject unsupported parameters.

This is a required ethtool op since kernel 5.7.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko committed:
ipq40xx: 5.10: move AR40xx to MDIO drivers

MDIO drivers were moved into their own sub directory of networking drivers.
This has caused the AR40xx driver to probe before MDIO drivers and that wont
work as it depends on the MDIO bus to be up so it can be fetched.

Lets solve it by moving the AR40xx into MDIO folder so they get probed like
before.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko commented:
@PolynomialDivision Here it is, It would be great if you can do some testing so we can get this merged

PolynomialDivision commented:
> @PolynomialDivision Here it is, It would be great if you can do some testing so we can get this merged

Great. Build in progress.. :D Thanks a lot.

PolynomialDivision commented:
@robimarko  Fails for me on the ip40xx Fritz!Box 7530?

```
  LD      vmlinux
  SORTTAB vmlinux
  SYSMAP  System.map
  OBJCOPY arch/arm/boot/Image
  Kernel: arch/arm/boot/Image is ready
  LDS     arch/arm/boot/compressed/vmlinux.lds
  AS      arch/arm/boot/compressed/head.o
  XZKERN  arch/arm/boot/compressed/piggy_data
  AS      arch/arm/boot/compressed/piggy.o
  CC      arch/arm/boot/compressed/misc.o
  CC      arch/arm/boot/compressed/decompress.o
  CC      arch/arm/boot/compressed/string.o
  SHIPPED arch/arm/boot/compressed/hyp-stub.S
  AS      arch/arm/boot/compressed/hyp-stub.o
  SHIPPED arch/arm/boot/compressed/lib1funcs.S
  AS      arch/arm/boot/compressed/lib1funcs.o
  SHIPPED arch/arm/boot/compressed/ashldi3.S
  AS      arch/arm/boot/compressed/ashldi3.o
  SHIPPED arch/arm/boot/compressed/bswapsdi2.S
  AS      arch/arm/boot/compressed/bswapsdi2.o
  LD      arch/arm/boot/compressed/vmlinux
  OBJCOPY arch/arm/boot/zImage
  Kernel: arch/arm/boot/zImage is ready
make[4]: Nothing to be done for 'Image'.
make[5]: *** No rule to make target 'arch/arm/boot/dts/qcom-ipq4019-one.dtb', needed by '__build'.  Stop.
make[4]: *** [Makefile:1359: dtbs] Error 2
make[4]: Leaving directory '/home/nick/openwrt/build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/linux-ipq40xx_generic/linux-5.10.63'
make[3]: *** [Makefile:25: /home/nick/openwrt/build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/linux-ipq40xx_generic/linux-5.10.63/.modules] Error 2
make[3]: Leaving directory '/home/nick/openwrt/target/linux/ipq40xx'
make[2]: *** [Makefile:11: compile] Error 2
make[2]: Leaving directory '/home/nick/openwrt/target/linux'
time: target/linux/compile#688.36#66.01#780.25
    ERROR: target/linux failed to build.
make[1]: *** [target/Makefile:25: target/linux/compile] Error 1
make[1]: Leaving directory '/home/nick/openwrt'
make: *** [/home/nick/openwrt/include/toplevel.mk:230: target/linux/compile] Error 2
```

robimarko commented:
Yeah, sorry I left the currently not upstream DTB to be built when I was testing in the patch.
Give me a minute to drop it.

robimarko committed:
ipq40xx: 5.10: copy patches

Copy over the 5.4 kernel patches to 5.10 folder.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

robimarko commented:
@PolynomialDivision Fixed, try now

PolynomialDivision commented:
LGTM to me:
https://hopglass.berlin.freifunk.net/#!v:m;n:schreiner-core.olsr

http://monitor.berlin.freifunk.net/host.php?h=schreiner-core

I can let that run a bit in our mesh network. But I don't see any errors and iperf3 measurements are also fine. :)

Boos4721 commented:
humm DTB seems to be missing
Devices: ``P2W R619AC``
```
cmd_syscalls=  KERNELRELEASE=5.10.64 CC="ccache_cc" zImage Image dtbs modules
4752
make[5]: Entering directory '/home/runner/work/openwrt/openwrt/build_dir/target-arm_cortex-a7+neon-vfpv4_musl_eabi/linux-ipq40xx/linux-5.10.64'
4753
make[6]: *** No rule to make target 'arch/arm/boot/dts/qcom-ipq4018-magic-2-wifi-next.dtb', needed by '__build'.  Stop.
4754
make[5]: *** [Makefile:1359: dtbs] Error 2

```

robimarko commented:
Are you doing something custom as there is a patch to add the qcom-ipq4018-magic-2-wifi-next.dtb to Makefile and it builds fine for me?

Boos4721 commented:
> Are you doing something custom as there is a patch to add the qcom-ipq4018-magic-2-wifi-next.dtb to Makefile and it builds fine for me?

idk, u can check [it](https://github.com/coolsnowwolf/lede/blob/master/target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-r619ac.dtsi).

robimarko commented:
But that DTS doesn't exist in OpenWrt at all, what does your tree have to do with this PR?

xabolcs commented:
@PolynomialDivision , how do you pack k510 kmods, how do you build the IB?

I'm too noob to use the manually built IB, as its finds  k54 dependencies for some kmods.
Sure, I'm doing something wrong.

So, how does your workflow looks?

I'd like to flash a GL-B1300 through ImageBuilder with some custom package.
IB builds nicely but creating an image with it, dies with k54 dependency error. 🙈

PolynomialDivision commented:
Maybe cleanup your buildroot? I just select all needed packages and that I want to build the testing kernel. And then I build the imagebuilder.

PolynomialDivision commented:
Also tested on **Mikrotik SxtSq AC 5**. Compiles and runs without any issues.

robimarko commented:
> > Are you doing something custom as there is a patch to add the qcom-ipq4018-magic-2-wifi-next.dtb to Makefile and it builds fine for me?
> 
> Do you mean me? No. I'm just compiling the imagebuilder for the 7530. Afterwards I use the imagbeuilder with some custom packages to create a mesh router. However, the modifications are just some userspace programs modifying luci and adding statistics. Everything compiles and runs! :)
> 
> Now only DSA is missing. ;)
> 
> Summarizing: Approved!

Not you, there was a question from somebody for an out of tree device and why the DTB was not built, but it looks like they deleted the question.

PolynomialDivision commented:
Can we speed up things a little bit and merge this, since it is only about kernel testing version? When I will further test that change and hopefully we will bump standard kernel version to 5.10. :)

pkgadd commented:
I've just successfully tested this PR on my ASUS 'lyra' MAP-AC2200. The only regression I've noticed so far is with its TI/National LP5523/55231 LED driver chip, which doesn't seem to initialize anymore `lp5523x: probe of 0-0032 failed with error -22`, leaving the rgb LED in its default state (cycling through all colours in rainbow mode).

```
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.10.64 (slh@trident) (arm-openwrt-linux-muslgnueabi-gcc (OpenWrt GCC 10.3.0 r17520-5ef4608c02) 10.3.0, GNU ld (GNU Binutils) 2.36.1) #0 SMP Tue Sep 14 20:42:38 2021
[    0.000000] CPU: ARMv7 Processor [410fc075] revision 5 (ARMv7), cr=10c5387d
[    0.000000] CPU: div instructions available: patching division code
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: ASUS Lyra MAP-AC2200
[    0.000000] Memory policy: Data cache writealloc
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000080000000-0x000000008fffffff]
[    0.000000]   HighMem  empty
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000080000000-0x0000000087dfffff]
[    0.000000]   node   0: [mem 0x0000000087e00000-0x0000000087ffffff]
[    0.000000]   node   0: [mem 0x0000000088000000-0x000000008fffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000080000000-0x000000008fffffff]
[    0.000000] On node 0 totalpages: 65536
[    0.000000]   Normal zone: 576 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 65536 pages, LIFO batch:15
[    0.000000] percpu: Embedded 15 pages/cpu s31244 r8192 d22004 u61440
[    0.000000] pcpu-alloc: s31244 r8192 d22004 u61440 alloc=15*4096
[    0.000000] pcpu-alloc: [0] 0 [0] 1 [0] 2 [0] 3 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 64960
[    0.000000] Kernel command line: root_rfs=0x00000000 flash_type=nand 
[    0.000000] Dentry cache hash table entries: 32768 (order: 5, 131072 bytes, linear)
[    0.000000] Inode-cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 247184K/262144K available (6222K kernel code, 600K rwdata, 1560K rodata, 1024K init, 238K bss, 14960K reserved, 0K cma-reserved, 0K highmem)
[    0.000000] SLUB: HWalign=64, Order=0-3, MinObjects=0, CPUs=4, Nodes=1
[    0.000000] rcu: Hierarchical RCU implementation.
[    0.000000]  Tracing variant of Tasks RCU enabled.
[    0.000000] rcu: RCU calculated value of scheduler-enlistment delay is 100 jiffies.
[    0.000000] NR_IRQS: 16, nr_irqs: 16, preallocated irqs: 16
[    0.000000] random: get_random_bytes called from start_kernel+0x360/0x508 with crng_init=0
[    0.000000] arch_timer: cp15 timer(s) running at 48.00MHz (virt).
[    0.000000] clocksource: arch_sys_counter: mask: 0xffffffffffffff max_cycles: 0xb11fd3bfb, max_idle_ns: 440795203732 ns
[    0.000008] sched_clock: 56 bits at 48MHz, resolution 20ns, wraps every 4398046511096ns
[    0.000023] Switching to timer-based delay loop, resolution 20ns
[    0.000234] Calibrating delay loop (skipped), value calculated using timer frequency.. 96.00 BogoMIPS (lpj=48000)
[    0.000256] pid_max: default: 32768 minimum: 301
[    0.000431] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.000449] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.001485] CPU: Testing write buffer coherency: ok
[    0.001771] qcom_scm: convention: smc legacy
[    0.002548] Setting up static identity map for 0x80300000 - 0x8030003c
[    0.002680] rcu: Hierarchical SRCU implementation.
[    0.002888] dyndbg: Ignore empty _ddebug table in a CONFIG_DYNAMIC_DEBUG_CORE build
[    0.003201] smp: Bringing up secondary CPUs ...
[    0.006208] smp: Brought up 1 node, 4 CPUs
[    0.006234] SMP: Total of 4 processors activated (384.00 BogoMIPS).
[    0.006245] CPU: All CPU(s) started in SVC mode.
[    0.011048] VFP support v0.3: implementor 41 architecture 2 part 30 variant 7 rev 5
[    0.011241] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 1911260446275000 ns
[    0.011269] futex hash table entries: 1024 (order: 4, 65536 bytes, linear)
[    0.011506] pinctrl core: initialized pinctrl subsystem
[    0.012718] NET: Registered protocol family 16
[    0.013057] DMA: preallocated 256 KiB pool for atomic coherent allocations
[    0.014227] thermal_sys: Registered thermal governor 'step_wise'
[    0.014620] cpuidle: using governor ladder
[    0.014674] cpuidle: using governor menu
[    0.033205] gpio-456 (enable external PA output-high): hogged as output/high
[    0.033246] gpio-458 (enable external PA output-high): hogged as output/high
[    0.033273] gpio-457 (enable external PA output-low): hogged as output/low
[    0.033298] gpio-459 (enable external PA output-low): hogged as output/low
[    0.040903] cryptd: max_cpu_qlen set to 1000
[    0.044685] usbcore: registered new interface driver usbfs
[    0.044758] usbcore: registered new interface driver hub
[    0.044817] usbcore: registered new device driver usb
[    0.044915] pps_core: LinuxPPS API ver. 1 registered
[    0.044929] pps_core: Software ver. 5.3.6 - Copyright 2005-2007 Rodolfo Giometti <giometti@linux.it>
[    0.044962] PTP clock support registered
[    0.046662] clocksource: Switched to clocksource arch_sys_counter
[    0.047715] NET: Registered protocol family 2
[    0.047872] IP idents hash table entries: 4096 (order: 3, 32768 bytes, linear)
[    0.048527] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 6144 bytes, linear)
[    0.048583] TCP established hash table entries: 2048 (order: 1, 8192 bytes, linear)
[    0.048625] TCP bind hash table entries: 2048 (order: 2, 16384 bytes, linear)
[    0.048714] TCP: Hash tables configured (established 2048 bind 2048)
[    0.048846] UDP hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.048886] UDP-Lite hash table entries: 256 (order: 1, 8192 bytes, linear)
[    0.049138] NET: Registered protocol family 1
[    0.049182] PCI: CLS 0 bytes, default 64
[    0.051426] workingset: timestamp_bits=14 max_order=16 bucket_order=2
[    0.056115] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.056141] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.192110] qcom-pcie 40000000.pci: host bridge /soc/pci@40000000 ranges:
[    0.192170] qcom-pcie 40000000.pci:       IO 0x0040200000..0x00402fffff -> 0x0040200000
[    0.192201] qcom-pcie 40000000.pci:      MEM 0x0040300000..0x0040ffffff -> 0x0040300000
[    0.463236] qcom-pcie 40000000.pci: Link up
[    0.463393] qcom-pcie 40000000.pci: PCI host bridge to bus 0000:00
[    0.463418] pci_bus 0000:00: root bus resource [bus 00-ff]
[    0.463436] pci_bus 0000:00: root bus resource [io  0x0000-0xfffff] (bus address [0x40200000-0x402fffff])
[    0.463452] pci_bus 0000:00: root bus resource [mem 0x40300000-0x40ffffff]
[    0.463503] pci 0000:00:00.0: [17cb:1001] type 01 class 0x060400
[    0.463532] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x00000fff]
[    0.463604] pci 0000:00:00.0: PME# supported from D0 D3hot
[    0.465183] PCI: bus0: Fast back to back transfers disabled
[    0.465522] pci 0000:01:00.0: [168c:0056] type 00 class 0x028000
[    0.465726] pci 0000:01:00.0: reg 0x10: [mem 0x00000000-0x001fffff 64bit]
[    0.466564] pci 0000:01:00.0: PME# supported from D0 D3hot
[    0.468332] PCI: bus1: Fast back to back transfers disabled
[    0.468385] pci 0000:00:00.0: BAR 8: assigned [mem 0x40400000-0x405fffff]
[    0.468403] pci 0000:00:00.0: BAR 0: assigned [mem 0x40300000-0x40300fff]
[    0.468427] pci 0000:01:00.0: BAR 0: assigned [mem 0x40400000-0x405fffff 64bit]
[    0.468518] pci 0000:00:00.0: PCI bridge to [bus 01-ff]
[    0.468535] pci 0000:00:00.0:   bridge window [mem 0x40400000-0x405fffff]
[    0.471450] pcieport 0000:00:00.0: AER: enabled with IRQ 103
[    0.473561] bam-dma-engine 8e04000.dma: num-channels unspecified in dt
[    0.473585] bam-dma-engine 8e04000.dma: num-ees unspecified in dt
[    0.475039] tcsr 1949000.tcsr: setting wifi_glb_cfg = 41000000
[    0.475142] tcsr 1953000.ess_tcsr: setting ess interface select = 0
[    0.475235] tcsr 1957000.tcsr: setting wifi_noc_memtype_m0_m2 = 2222222
[    0.475550] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    0.476158] msm_serial 78af000.serial: msm_serial: detected port #0
[    0.476203] msm_serial 78af000.serial: uartclk = 1843200
[    0.476256] 78af000.serial: ttyMSM0 at MMIO 0x78af000 (irq = 33, base_baud = 115200) is a MSM
[    0.476286] msm_serial: console setup on port #0
[    1.182658] printk: console [ttyMSM0] enabled
[    1.187824] msm_serial: driver initialized
[    1.196881] loop: module loaded
[    1.198525] nand: device found, Manufacturer ID: 0xc2, Chip ID: 0xf1
[    1.198875] nand: Macronix MX30LF1G18AC
[    1.205449] nand: 128 MiB, SLC, erase size: 128 KiB, page size: 2048, OOB size: 64
[    1.209016] Block protection check failed
[    1.216723] 7 fixed-partitions partitions found on MTD device qcom_nand.0
[    1.220732] Creating 7 MTD partitions on "qcom_nand.0":
[    1.227504] 0x000000000000-0x000000080000 : "SBL1"
[    1.233688] 0x000000080000-0x000000100000 : "MIBIB"
[    1.238448] 0x000000100000-0x000000200000 : "QSEE"
[    1.243699] 0x000000200000-0x000000280000 : "CDT"
[    1.248060] 0x000000280000-0x0000003c0000 : "APPSBL"
[    1.253688] 0x0000003c0000-0x000000400000 : "APPSBLENV"
[    1.257609] 0x000000400000-0x000008000000 : "ubi"
[    1.272124] random: fast init done
[    1.373843] libphy: Fixed MDIO Bus: probed
[    1.374677] libphy: ipq4019_mdio: probed
[    1.408279] ESS reset ok!
[    1.432487] ESS reset ok!
[    1.858464] EDMA using MAC@ - using
[    1.858482] 86:93:03:9d:82:1d
[    1.861683] EDMA using MAC@ - using
[    1.861697] 5e:89:64:d9:ed:01
[    1.927845] i2c /dev entries driver
[    1.928109] i2c_qup 78b7000.i2c: using default clock-frequency 100000
[    1.933003] sdhci: Secure Digital Host Controller Interface driver
[    1.936774] sdhci: Copyright(c) Pierre Ossman
[    1.942832] sdhci-pltfm: SDHCI platform and OF driver helper
[    1.947521] lp5523x: probe of 0-0032 failed with error -22
[    1.956535] NET: Registered protocol family 10
[    1.959702] Segment Routing with IPv6
[    1.962807] NET: Registered protocol family 17
[    1.966516] 8021q: 802.1Q VLAN Support v1.8
[    1.970879] Registering SWP/SWPB emulation handler
[    1.986091] UBI: auto-attach mtd6
[    1.986136] ubi0: attaching mtd6
[    2.864282] ubi0: scanning is finished
[    2.879937] ubi0: attached mtd6 (name "ubi", size 124 MiB)
[    2.879986] ubi0: PEB size: 131072 bytes (128 KiB), LEB size: 126976 bytes
[    2.884383] ubi0: min./max. I/O unit sizes: 2048/2048, sub-page size 2048
[    2.891236] ubi0: VID header offset: 2048 (aligned 2048), data offset: 4096
[    2.898077] ubi0: good PEBs: 991, bad PEBs: 1, corrupted PEBs: 0
[    2.904843] ubi0: user volume: 8, internal volumes: 1, max. volumes count: 128
[    2.911096] ubi0: max/mean erase counter: 241/88, WL threshold: 4096, image sequence number: 2108563064
[    2.918131] ubi0: available PEBs: 0, total reserved PEBs: 991, PEBs reserved for bad PEB handling: 19
[    2.927460] ubi0: background thread "ubi_bgt0d" started, PID 117
[    2.937765] block ubiblock0_6: created from ubi0:6(rootfs)
[    2.942877] ubiblock: device ubiblock0_6 (rootfs) set to be root filesystem
[    2.954551] VFS: Mounted root (squashfs filesystem) readonly on device 254:0.
[    2.956580] Freeing unused kernel memory: 1024K
[    2.960964] Run /sbin/init as init process
[    2.965031]   with arguments:
[    2.965039]     /sbin/init
[    2.965044]   with environment:
[    2.965049]     HOME=/
[    2.965054]     TERM=linux
[    2.965059]     root_rfs=0x00000000
[    2.965063]     flash_type=nand
[    3.189359] random: crng init done
[    3.843486] init: Console is alive
[    3.843845] init: - watchdog -
[    6.871866] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    6.980245] genirq: irq_chip msmgpio did not update eff. affinity mask of irq 104
[    6.982612] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    6.991201] SCSI subsystem initialized
[    6.998692] ehci-fsl: Freescale EHCI Host controller driver
[    6.999872] ehci-platform: EHCI generic platform driver
[    7.006884] dwc3-qcom 60f8800.usb2: IRQ hs_phy_irq not found
[    7.008774] dwc3-qcom 60f8800.usb2: IRQ dp_hs_phy_irq not found
[    7.014666] dwc3-qcom 60f8800.usb2: IRQ dm_hs_phy_irq not found
[    7.020302] dwc3-qcom 60f8800.usb2: IRQ ss_phy_irq not found
[    7.081352] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    7.081427] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 1
[    7.086024] xhci-hcd xhci-hcd.0.auto: hcc params 0x0220f665 hci version 0x100 quirks 0x0000000002010010
[    7.093405] xhci-hcd xhci-hcd.0.auto: irq 106, io mem 0x06000000
[    7.103891] hub 1-0:1.0: USB hub found
[    7.109046] hub 1-0:1.0: 1 port detected
[    7.113476] xhci-hcd xhci-hcd.0.auto: xHCI Host Controller
[    7.117301] xhci-hcd xhci-hcd.0.auto: new USB bus registered, assigned bus number 2
[    7.122589] xhci-hcd xhci-hcd.0.auto: Host supports USB 3.0 SuperSpeed
[    7.130254] usb usb2: We don't know the algorithms for LPM for this host, disabling LPM.
[    7.137392] hub 2-0:1.0: USB hub found
[    7.145038] hub 2-0:1.0: config failed, hub doesn't have any ports! (err -19)
[    7.153340] usbcore: registered new interface driver usb-storage
[    7.160796] usbcore: registered new interface driver uas
[    7.163056] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    7.175309] init: - preinit -
[    7.355719] usb 1-1: new full-speed USB device number 2 using xhci-hcd
[   10.007804] ess_edma c080000.edma eth1: Link is Down
[   12.068601] ess_edma c080000.edma eth1: Link is Up - 100Mbps/Full - flow control rx/tx
[   12.070320] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
[   20.438988] mount_root: loading kmods from internal overlay
[   20.462030] kmodloader: loading kernel modules from //etc/modules-boot.d/*
[   20.464806] kmodloader: done loading kernel modules from //etc/modules-boot.d/*
[   21.081398] UBIFS (ubi0:7): Mounting in unauthenticated mode
[   21.081658] UBIFS (ubi0:7): background thread "ubifs_bgt0_7" started, PID 245
[   21.195437] UBIFS (ubi0:7): UBIFS: mounted UBI device 0, volume 7, name "rootfs_data"
[   21.195538] UBIFS (ubi0:7): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[   21.202468] UBIFS (ubi0:7): FS size: 36823040 bytes (35 MiB, 290 LEBs), journal size 1904640 bytes (1 MiB, 15 LEBs)
[   21.212276] UBIFS (ubi0:7): reserved for root: 1739243 bytes (1698 KiB)
[   21.222521] UBIFS (ubi0:7): media format: w5/r0 (latest is w5/r0), UUID 7B6ECB01-B6A2-4A82-BA1C-CA4F466582C2, small LPT model
[   21.230757] block: attempting to load /tmp/ubifs_cfg/upper/etc/config/fstab
[   21.240861] block: unable to load configuration (fstab: Entry not found)
[   21.247435] block: attempting to load /tmp/ubifs_cfg/etc/config/fstab
[   21.254415] block: unable to load configuration (fstab: Entry not found)
[   21.260700] block: attempting to load /etc/config/fstab
[   21.278278] block: unable to load configuration (fstab: Entry not found)
[   21.278517] block: no usable configuration
[   21.288101] UBIFS (ubi0:7): un-mount UBI device 0
[   21.288248] UBIFS (ubi0:7): background thread "ubifs_bgt0_7" stops
[   21.299528] UBIFS (ubi0:7): Mounting in unauthenticated mode
[   21.300486] UBIFS (ubi0:7): background thread "ubifs_bgt0_7" started, PID 246
[   21.411239] UBIFS (ubi0:7): UBIFS: mounted UBI device 0, volume 7, name "rootfs_data"
[   21.411340] UBIFS (ubi0:7): LEB size: 126976 bytes (124 KiB), min./max. I/O unit sizes: 2048 bytes/2048 bytes
[   21.418265] UBIFS (ubi0:7): FS size: 36823040 bytes (35 MiB, 290 LEBs), journal size 1904640 bytes (1 MiB, 15 LEBs)
[   21.428082] UBIFS (ubi0:7): reserved for root: 1739243 bytes (1698 KiB)
[   21.438321] UBIFS (ubi0:7): media format: w5/r0 (latest is w5/r0), UUID 7B6ECB01-B6A2-4A82-BA1C-CA4F466582C2, small LPT model
[   21.672519] block: attempting to load /tmp/ubifs_cfg/upper/etc/config/fstab
[   21.672808] block: unable to load configuration (fstab: Entry not found)
[   21.678595] block: attempting to load /tmp/ubifs_cfg/etc/config/fstab
[   21.685524] block: unable to load configuration (fstab: Entry not found)
[   21.691817] block: attempting to load /etc/config/fstab
[   21.698600] block: unable to load configuration (fstab: Entry not found)
[   21.703435] block: no usable configuration
[   21.715957] mount_root: overlay filesystem has not been fully initialized yet
[   21.717036] mount_root: switching to ubifs overlay
[   34.940085] urandom-seed: Seed file not found (/etc/urandom.seed)
[   35.031244] ess_edma c080000.edma eth1: Link is Down
[   35.039216] procd: - early -
[   35.039421] procd: - watchdog -
[   35.591990] procd: - watchdog -
[   35.592604] procd: - ubus -
[   35.830783] procd: - init -
[   36.741221] kmodloader: loading kernel modules from /etc/modules.d/*
[   36.754131] urngd: v1.0.2 started.
[   36.770238] sit: IPv6, IPv4 and MPLS over IPv4 tunneling driver
[   36.848590] hid: raw HID events driver (C) Jiri Kosina
[   36.851355] GACT probability on
[   36.853803] Mirror/redirect action on
[   36.865038] u32 classifier
[   36.865073]     input device check on
[   36.866668]     Actions configured
[   36.889151] fuse: init (API version 7.32)
[   36.910823] Bluetooth: Core ver 2.22
[   36.910963] NET: Registered protocol family 31
[   36.913546] Bluetooth: HCI device and connection manager initialized
[   36.917814] Bluetooth: HCI socket layer initialized
[   36.924309] Bluetooth: L2CAP socket layer initialized
[   36.928932] Bluetooth: SCO socket layer initialized
[   36.935762] Bluetooth: BNEP (Ethernet Emulation) ver 1.3
[   36.940845] Bluetooth: BNEP filters: protocol multicast
[   36.947621] Bluetooth: BNEP socket layer initialized
[   36.951967] usbcore: registered new interface driver btusb
[   36.958135] Loading modules backported from Linux version v5.10.64-0-gcb83afdc0b86
[   36.959857] Backport generated by backports.git v5.10.42-1-0-gbee5c545
[   36.964243] Bluetooth: HCI UART driver ver 2.3
[   36.969035] Bluetooth: HCI UART protocol H4 registered
[   36.974767] Bluetooth: HCI UART protocol BCSP registered
[   36.974798] Bluetooth: HCI UART protocol ATH3K registered
[   36.981731] Bluetooth: HIDP (Human Interface Emulation) ver 1.2
[   36.984672] Bluetooth: HIDP socket layer initialized
[   37.013747] Bluetooth: RFCOMM TTY layer initialized
[   37.013834] Bluetooth: RFCOMM socket layer initialized
[   37.017533] Bluetooth: RFCOMM ver 1.11
[   37.025477] usbcore: registered new interface driver usblp
[   37.038691] xt_time: kernel timezone is -0000
[   37.040673] usbcore: registered new interface driver ath3k
[   37.134548] PPP generic driver version 2.4.2
[   37.136362] NET: Registered protocol family 24
[   37.141714] wireguard: WireGuard 1.0.0 loaded. See www.wireguard.com for information.
[   37.142216] wireguard: Copyright (C) 2015-2019 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
[   37.180523] ath10k_pci 0000:01:00.0: enabling device (0140 -> 0142)
[   37.181382] ath10k_pci 0000:01:00.0: pci irq msi oper_irq_mode 2 irq_mode 0 reset_mode 0
[   40.683626] ath10k_pci 0000:01:00.0: qca9888 hw2.0 target 0x01000000 chip_id 0x00000000 sub 0000:0000
[   40.683709] ath10k_pci 0000:01:00.0: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
[   40.696730] ath10k_pci 0000:01:00.0: firmware ver 10.4-3.9.0.2-00152 api 5 features no-p2p,mfp,peer-flow-ctrl,allows-mesh-bcast,no-ps crc32 ddd53b82
[   41.066524] ath10k_pci 0000:01:00.0: board_file api 2 bmi_id 0:17 crc32 f228337a
[   42.973347] ath10k_pci 0000:01:00.0: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 512 raw 0 hwcrypto 1
[   43.088559] ath: EEPROM regdomain sanitized
[   43.088580] ath: EEPROM regdomain: 0x64
[   43.088586] ath: EEPROM indicates we should expect a direct regpair map
[   43.088600] ath: Country alpha2 being used: 00
[   43.088606] ath: Regpair used: 0x64
[   44.242905] ath10k_ahb a000000.wifi: qca4019 hw1.0 target 0x01000000 chip_id 0x003b00ff sub 0000:0000
[   44.242957] ath10k_ahb a000000.wifi: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
[   44.254757] ath10k_ahb a000000.wifi: firmware ver 10.4-3.2.1.1-00035 api 5 features no-p2p,raw-mode,mfp,peer-flow-ctrl,btcoex-param,allows-mesh-bcast,no-ps crc32 29dcf3a6
[   44.334034] ath10k_ahb a000000.wifi: board_file api 2 bmi_id 0:20 crc32 e2dfaa91
[   45.658250] ath10k_ahb a000000.wifi: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 512 raw 0 hwcrypto 1
[   45.672890] ath: EEPROM regdomain sanitized
[   45.672908] ath: EEPROM regdomain: 0x64
[   45.672914] ath: EEPROM indicates we should expect a direct regpair map
[   45.672929] ath: Country alpha2 being used: 00
[   45.672934] ath: Regpair used: 0x64
[   45.978951] ath10k_ahb a800000.wifi: qca4019 hw1.0 target 0x01000000 chip_id 0x003b00ff sub 0000:0000
[   45.979019] ath10k_ahb a800000.wifi: kconfig debug 0 debugfs 1 tracing 0 dfs 1 testmode 0
[   45.992356] ath10k_ahb a800000.wifi: firmware ver 10.4-3.2.1.1-00035 api 5 features no-p2p,raw-mode,mfp,peer-flow-ctrl,btcoex-param,allows-mesh-bcast,no-ps crc32 29dcf3a6
[   46.040196] ath10k_ahb a800000.wifi: board_file api 2 bmi_id 0:21 crc32 e2dfaa91
[   47.385536] ath10k_ahb a800000.wifi: htt-ver 2.2 wmi-op 6 htt-op 4 cal pre-cal-file max-sta 512 raw 0 hwcrypto 1
[   47.452269] ath: EEPROM regdomain sanitized
[   47.452290] ath: EEPROM regdomain: 0x64
[   47.452296] ath: EEPROM indicates we should expect a direct regpair map
[   47.452310] ath: Country alpha2 being used: 00
[   47.452315] ath: Regpair used: 0x64
[   47.463904] kmodloader: done loading kernel modules from /etc/modules.d/*
[   65.048436] br-lan: port 1(eth1) entered blocking state
[   65.048511] br-lan: port 1(eth1) entered disabled state
[   65.053164] device eth1 entered promiscuous mode
[   66.017750] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[   67.107165] ess_edma c080000.edma eth1: Link is Up - 100Mbps/Full - flow control rx/tx
[   68.201449] br-lan: port 1(eth1) entered blocking state
[   68.201513] br-lan: port 1(eth1) entered forwarding state
[   68.208773] br-lan: port 2(wlan1) entered blocking state
[   68.211100] br-lan: port 2(wlan1) entered disabled state
[   68.216895] device wlan1 entered promiscuous mode
[   68.222924] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready
[   70.705893] ath10k_pci 0000:01:00.0: Unknown eventid: 36933
[   70.774269] br-lan: port 3(wlan0) entered blocking state
[   70.774317] br-lan: port 3(wlan0) entered disabled state
[   70.779146] device wlan0 entered promiscuous mode
[   70.784303] br-lan: port 3(wlan0) entered blocking state
[   70.788622] br-lan: port 3(wlan0) entered forwarding state
[   70.796388] br-lan: port 3(wlan0) entered disabled state
[   72.349712] br-lan: port 4(wlan2) entered blocking state
[   72.349768] br-lan: port 4(wlan2) entered disabled state
[   72.354576] device wlan2 entered promiscuous mode
[   72.716467] IPv6: ADDRCONF(NETDEV_CHANGE): wlan0: link becomes ready
[   72.717099] br-lan: port 3(wlan0) entered blocking state
[   72.721991] br-lan: port 3(wlan0) entered forwarding state
[   73.241445] IPv6: ADDRCONF(NETDEV_CHANGE): wlan2: link becomes ready
[   73.241912] br-lan: port 4(wlan2) entered blocking state
[   73.246928] br-lan: port 4(wlan2) entered forwarding state
[   83.463630] IPv6: ADDRCONF(NETDEV_CHANGE): wlan1: link becomes ready
[   83.464062] br-lan: port 2(wlan1) entered blocking state
[   83.469213] br-lan: port 2(wlan1) entered forwarding state
```

pkgadd commented:
Some further debugging has revealed that "[leds: lp55xx: Initialize enable GPIO direction to output](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a5d3d1adc95f4ac5968b7a77ee95a3abbbb96f49)" is missing from kernel v5.10, I've quickly added it as `target/linux/generic/backport-5.10/075-leds_lp55xx_initialize-enable-GPIO-direction-to.output.patch` and the LP5523 controller is working again (initializing the LEDs to off (thank goodness, spares me from the Y.M.C.A.) and allowing manual colour selection via the LED subsystem).

```
[    1.873236] lp5523x 0-0032: lp5523 Programmable led chip found
```

With "[leds: lp55xx: Initialize enable GPIO direction to output](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=a5d3d1adc95f4ac5968b7a77ee95a3abbbb96f49)" added:
```
Tested-by: Stefan Lippers-Hollmann <s.l-h@gmx.de> [ipq4019/ map-ac2200]
```

adschm commented:
Nicely done. Let's collect a few other Tested-by and then get it in ...

robimarko commented:
@pkgadd Can you send that backport as a patch or PR to add it to generic target?

PolynomialDivision commented:
> Nicely done. Let's collect a few other Tested-by and then get it in ...

Tested-by: Nick Hainke <vincent@systemli.org> [ipq4019/fritzbox-7530, ipq4019/sxtsq-5ac]

adschm commented:
@pkgadd @PolynomialDivision Would you please re-test with LEDs patch?

pkgadd commented:
> @pkgadd […] Would you please re-test with LEDs patch?

hmm, I really have no idea why, but I'm back up `lp5523x: probe of 0-0032 failed with error -22` with current [HEAD of this PR](https://github.com/sartura/openwrt/commit/e9166ed70e40b4e105bc2ee05f19652ea73e1bb7) merged into [master](https://github.com/openwrt/openwrt/commit/a46fa5c3a7). I honestly have no idea how, or why at the moment - the patch is clearly there and applied correctly (confirmed that) and yesterday's build was working over several reboots and a full power-down over night (I flashed back yesterday's build based on master's g5ef4608c02 and it's immediately working again). Patch and general functionality is fine though, it's just the (very annoying LED cycling to all colours of the rainbow continuously), I'll obviously need to apply further debugging.

luaraneda commented:
If you do end applying the LED patch please rename the file to include the Kernel release as other backport patches do.

From a quick search it was first applied for 5.14, so the name would look like:
830-**v5.14**-leds-lp55xx-Initialize-enable-GPIO-direction-to-outp.patch

robimarko commented:
Sorry, I completely forgot about that after exporting.

robimarko committed:
kernel: 5.10: backport fix for lp55xx LED driver

This backports the upstream commit:
leds: lp55xx: Initialize enable GPIO direction to output

Without it under kernel 5.10 on Asus MAP-AC2200
the LED driver will fail probing:
[    1.947521] lp5523x: probe of 0-0032 failed with error -22

After the backported fix:
[    1.873236] lp5523x 0-0032: lp5523 Programmable led chip found

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

luaraneda commented:
That was a really quick response time.
Thanks!

PolynomialDivision commented:
> @pkgadd @PolynomialDivision Would you please re-test with LEDs patch?

`Tested-by: Nick Hainke <vincent@systemli.org> [ipq4019/fritzbox-7530, ipq4019/sxtsq-5ac]`

Done. :)

PolynomialDivision commented:
Also tested on Fritz!Box 4040:
`Tested-by: Nick Hainke <vincent@systemli.org> [ipq4019/fritzbox-7530, [ipq4019/fritzbox-4040, ipq4019/sxtsq-5ac]`

kornel-swierzy commented:
Works on _8devices Habanero DVK_ :) [8devicesHabaneroDVK_dmesg.txt](https://github.com/openwrt/openwrt/files/7192854/8devicesHabaneroDVK_dmesg.txt)

PolynomialDivision commented:
What is still blocking this PR from being merged?

adschm commented:
So, we have three test reports, of which one has working LEDs, one didn't test, and one has broken LEDs?

PolynomialDivision commented:
I looked at the LEDs of the 4040. It was working. However, this backported patch only affects some other devices? I did not see any errors regarding `lp55xx` on the devices I tested.

robimarko commented:
@adschm LED-s are working fine in general, it looks to affect only the lp55xx driver, but since there are only 2 boards that use it and I don't have any of them I really cant do anything.
@pkgadd Was the one that reported the issue

PolynomialDivision commented:
It is only about the testing kernel version. I guess it is not the last issue we will experience. The devices won't brick and will still work totally fine. With this PR we can enable a broader pool of test-persons already going for testing kernel. So, I would like to see this PR merged. ;) I would also like to go for further testing.

xabolcs commented:
Complile tested, run tested.
Works nicely as an 5G wired AP, Snapcast and MPD server.
```
root@GL-B1300-078 ~ # uptime 
 18:07:19 up 53 min,  load average: 0.58, 0.83, 0.75
root@GL-B1300-078 ~ # dmesg |head
[    0.000000] Booting Linux on physical CPU 0x0
[    0.000000] Linux version 5.10.64 (xabolcs@ut2004) (arm-openwrt-linux-muslgnueabi-gcc (OpenWrt GCC 11.2.0 r17564+12-366be2183e) 11.2.0, GNU ld (GNU Binutils) 2.37) #0 SMP Thu Sep 23 06:05:06 2021
[    0.000000] CPU: ARMv7 Processor [410fc075] revision 5 (ARMv7), cr=10c5387d
[    0.000000] CPU: div instructions available: patching division code
[    0.000000] CPU: PIPT / VIPT nonaliasing data cache, VIPT aliasing instruction cache
[    0.000000] OF: fdt: Machine model: GL.iNet GL-B1300
[    0.000000] Memory policy: Data cache writealloc
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000080000000-0x000000008fffffff]
[    0.000000]   HighMem  empty
```


```
Tested-by: Szabolcs Hubai <szab.hu@gmail.com> [ipq4029/gl-b1300]
```

adschm commented:
Thanks, I will do a final build test and then merge it during the weekend.

PolynomialDivision commented:
Nice, thanks! :)

Maybe @robimarko can also publish the dsa draft for further testing and debugging? ;)
